### DISCUSSION TITLE
How to customise the plot tooltip?

### DISCUSSION THREAD

**Discussion by dulajra** [h2oai/wave#1832] [Q&A] [answered]
I am plotting a histogram and I have passed `x1=x_min`, `x2=x_max` and `y=y` values for the `ui.mark`. The plot is working fine but I see `x_min - x_max` in the plot tooltip in addition to  `x_min`, `x_max` and `y`. 

1. Aren't only the fields in the `data` included int the tooltip?
2. Is there a way to remove `x_min - x_max`  from the tooltip?

<img width="1436" alt="Screenshot 2023-02-13 at 22 08 48" src="https://user-images.githubusercontent.com/12434380/218517496-14bb87b7-0dfd-4bb6-ab97-9a4334457d18.png">

Sample code 

```
ui.visualization(
        plot=ui.plot(
            [
                ui.mark(
                    type="interval",
                    x='={{intl x_min type="time" dateStyle="short" timeStyle="short" }}',
                    x1='={{intl x_min type="time" dateStyle="short" timeStyle="short" }}',
                    x2='={{intl x_max type="time" dateStyle="short" timeStyle="short" }}',
                    x_scale="time",
                    x_title=histogram.x_label,
                    y_min=0,
                    y="=y",
                    y_title=histogram.y_label,
                    fill_color=SHOE_WAX,
                    stroke_color=IRON,
                    label_overlap="hide",
                )
            ]
        ),
        data=data(
            fields=["x_min", "x_max", "y"],
            pack=True,
            rows=d,
        ),
    )
```

**Comment by mturoci** [comment]
> Aren't only the fields in the data included int the tooltip?

Only fields in data included should be the correct behavior. The screenshot looks like a bug on our side - cc @marek-mihok for more info as the original feature author.

**Comment by dulajra** [comment]
And another unexpected behaviour. Both `x1` and `x2` has the timestamp and in the same format. Here are some sample data.

```
09 = {tuple: 3} ('2023-02-14T07:24', '2023-02-14T07:30', 50000)
10 = {tuple: 3} ('2023-02-14T07:30', '2023-02-14T07:35', 50000)
```

But on the tooltip the time part is not visible only for `x1`

<img width="1415" alt="Screenshot 2023-02-14 at 13 09 42" src="https://user-images.githubusercontent.com/12434380/218670631-ada9e970-9769-4fcc-92ae-420867c4710c.png">

**Comment by marek-mihok** [comment, marked as answer]
Since #1330 we display in the tooltip everything what is in the data. If the user passed 3 fields into the data but is plotting only 2 of them, all 3 fields are shown in the tooltip. However, in Wave we are refactoring the original data before plotting and for ranged (`type='interval'`) plots we are adding also the segment range into the data.

> Only fields in data included should be the correct behavior.

As @mturoci suggests, this may not be the correct bahavior and we could ignore this [ranged data](https://github.com/h2oai/wave/issues/1473) when rendering the tooltip. Feel free to [open the new issue](https://github.com/h2oai/wave/issues/new?assignees=&labels=bug&template=bug_report.md&title=) @dulajra . 

Regarding your [last comment](https://github.com/h2oai/wave/discussions/1832#discussioncomment-4967533), the time part should not be visible. Although I see where could be the issue, it will be "fixed" as well by not showing the ranged field.

**Reply by dulajra** [reply]
Bug reported https://github.com/h2oai/wave/issues/1836
